bellbot_action_server/CMakeLists.txt

```diff
@@ -126,41 +84,15 @@ include_directories(
 
 ## Mark executable scripts (Python etc.) for installation
 ## in contrast to setup.py, you can choose the destination
-# install(PROGRAMS
-#   scripts/my_python_script
-#   DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
-
-## Mark executables and/or libraries for installation
-# install(TARGETS bellbot_action_server bellbot_action_server_node
-#   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-#   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-#   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
-
-## Mark cpp header files for installation
-# install(DIRECTORY include/${PROJECT_NAME}/
-#   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-#   FILES_MATCHING PATTERN "*.h"
-#   PATTERN ".svn" EXCLUDE
-# )
-
-## Mark other files for installation (e.g. launch and bag files, etc.)
-# install(FILES
-#   # myfile1
-#   # myfile2
-#   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-# )
-
-#############
-## Testing ##
-#############
-
-## Add gtest based cpp test target and link libraries
-# catkin_add_gtest(${PROJECT_NAME}-test test/test_bellbot_action_server.cpp)
-# if(TARGET ${PROJECT_NAME}-test)
-#   target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
-# endif()
+install(PROGRAMS
+#  launch/bellbot_action_server.launch
+  src/bellbot_server.py
+  src/bellbot_state_machine.py
+  src/agent.py
+  src/bellbot_test_client.py
+  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
```

Comment:
Only `bellbot_server.py` and `bellbot_test_client.py` have a "main" function which implies that the other two are not meant to be executable. The correct way of installing these files would be to move `bellbot_server.py` and `bellbot_test_client.py` to a scripts directory and install them using this install target.

`agent.py` and `bellbot_state_machine.py` should be moved to `src/bellbot_action_server` and be installed via the `setup.py` because they shouldn't be executable and be available via the ROS python path. They can the be imported via `from bellbot_action_server.bellbot_state_machine import BellbotStateMachine` instead of `from bellbot_state_machine import BellbotStateMachine` which corresponds to a clearer structure: `from package.file import Class`.